libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -19,6 +19,72 @@ namespace eosio::chain {
    struct transaction_trace;
    using transaction_trace_ptr = std::shared_ptr<transaction_trace>;
 
+   // Identify the starting position of a sync call in the call's console log.
+   // This is used for pretty printing console logs.
+   struct console_marker {
+      uint32_t call_ordinal     = 0;
```

Comment:
When discussed on call I thought we came to the conclusion we only needed the position? i.e. type could be  `vector<unsigned_int> console_markers` and the size of `console_markers` is the number of sync calls performed I think[^1]. Can we still do it that way or was there a challenge with it?

[^1]: I seem to remember us saying number of sync calls minus 1 but that isn't making sense to me immediately :thinking:

Reply:
I thought using the struct was more clear. But will change to one dimensional vector to save space, as this is to be used by SHiP. `n-1` should be OK as we know the start and end.

Reply:
Hm, when I started to change the code, it seemed to me we need `n` instead of `n-1`. Did I miss something?

If we have `n` sync calls, we should have `n` points to indicate where in the `console` we start the calls.

If we make only 1 sync call, I think we still need 1marker, not 0.

Reply:
I think you are correct @linh2931.